runtime/startup_aux.c

```diff
@@ -171,6 +172,7 @@ CAMLexport void caml_shutdown(void)
   call_registered_value("Pervasives.do_at_exit");
   call_registered_value("Thread.at_shutdown");
```

Comment:
How are remaining memprof callbacks treated on program termination? Is there an effort somewhere to handle remaining callbacks? (This could be in `caml_finalise_heap` below?)

Reply:
Note: I'm not asking about this specifically in the `caml_shutdown` case, but in general in program-exit scenarios (I am not sure `caml_shutdown` is called on graceful process exit).

Because tracking is disabled within callbacks, in principle we could iterate on all remaining callbacks, ignoring exceptions¹, and know that we completely handled everything. It would seem natural to do this on program exit, and also on `caml_memprof_set`.

¹: re. exceptions, I vaguely remember that a decision was made on what to do for exceptions in program-exit scenarios (raise a warning?), the same policy could be reused when ignoring exceptions in disable-callbacks scenarios.

Reply:
We do nothing about callbacks on program exit. What we could do is registering in `Gc.ml` an `on_exit` handler which would call `Gc.Memprof.stop`, and hence call the remaining callbacks.

Reply:
Yes, I think it would be a nice way to ensure this. (With this approach you don't have to be careful about changing all end-of-execution points, they already call `do_at_exit`.). A mild inconvenient is that it means that some of the at_exit functions will be monitored and some (those running after this callback because they were inserted before) will not. I don't think we care to give guarantees about when monitoring will stop compared to other forms of callbacks running after the program-termination decision was made.